src/node/http.ts

```diff
@@ -502,9 +508,15 @@ export class HttpServer {
     })
   }
 
-  public dispose(): void {
+  /**
+   * Stop and dispose everything. Return an array of disposal errors.
+   */
+  public async dispose(): Promise<Error[]> {
     this.socketProvider.stop()
-    this.providers.forEach((p) => p.dispose())
+    const providers = Array.from(this.providers.values())
+    // Catch so all the errors can be seen rather than just the first one.
+    const responses = await Promise.all<Error | undefined>(providers.map((p) => p.dispose().catch((e) => e)))
+    return responses.filter<Error>((r): r is Error => typeof r !== "undefined")
```

Comment:
Why do we need to use `typeof` here versus `!== undefined`?

Reply:
Practically no difference, just a stylistic choice. It might be because I'm thinking of it in terms of the type rather than the value. It parallels what I'd do if I wanted to filter strings instead, and I like the consistency. It also works for undeclared variables, further enforcing the consistency.

It might also be partly force of habit from back when you could overwrite undefined on the global object. :stuck_out_tongue:

But even though the global undefined is read-only now it still feels weird to me that I can do `const undefined = "something"` and now my checks have a different meaning.